### DISCUSSION TITLE
Clamping MSS for IPsec packets based on machine MTU configured

### DISCUSSION THREAD

**Discussion by ashjosh1git** [strongswan/strongswan#1391] [Q&A]
**Issue --**
Even though the local MTU was set to 1500 on the Linux machine, the IPsec packets being sent on the wire were 1546 bytes long, which was resulting in to packets getting dropped on WAN. 

**Rule that worked --**
The following rule works but it is only optimized for 1500 MTU, whereas our systems in general supports configuring MTU to any of the 1500-9000+. So in effect, this rule is restricting our abilities to dynamically change MTUs with IPsec enabled, and in its current form, it clamps even higher MTU packets to 1360 MSS.

`iptables -t mangle -A INPUT -m policy --pol ipsec --dir in -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:65535 -j TCPMSS --set-mss 1360
`

**Query --** 
There is an iptables option _TCPMSS --clamp-mss-to-pmtu_. But we are not sure if that rule would take into account our system MTU size that we have set on that system, and clamp IPsec packets MSS to [(MTU SIZE - TCP/IP overhead (40 bytes) - Crypto overhead for tunnel mode with selected cipher (say up to 100 based on crypto algo we are using)].

There are also mentions rules like the following at many places, but it looks different than what has worked for us (i.e, INPUT vs FORWARD), so we are not sure if this is a right rule that can be used.

`iptables -t mangle -A FORWARD -m policy --pol ipsec --dir in -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:65535 -j TCPMSS --clamp-mss-to-pmtu
`

Basically, could you please suggest if there is any iptables (or nft) rule which considers the Linux system MTU size, and clamps IPsec packets MSS to :
[(MTU SIZE) - (TCP/IP overhead (40 bytes)) - (Crypto overhead for tunnel mode with selected cipher (say up to 100 based on crypto algo)]?

Thanks!

**Comment by Thermi** [comment]
Yes that's why you should use route based VPNs. You can configure that with IPsec on Linux by using an XFRM interface or a VTI. They can handle that. Or use routes with the set MTU for the remote network. Just make sure the route with the MTU is only used by the inner packet, not the outer packet. That could also work.

**Comment by ashjosh1git** [comment]
Sure, thanks.  We could use route based VPN approach as a long/medium term option. But as of now, is it possible to achieve such clamping with a rule based approach? If yes, could you help us with that please?

**Reply by Thermi** [reply]
There's no possibility to set the MTU with a rule. I personally have not had any success making PMTU discovery work over policy based IPsec. So clamping to pmtu will not work.
